gateways/adyen/request_builder.go

```diff
@@ -47,6 +75,18 @@ func buildAuthRequest(authRequest *sleet.AuthorizationRequest, merchantAccount s
 
 	addPaymentSpecificFields(authRequest, request)
 
+	// overwrites the flag transactions
+	if authRequest.ProcessingInitiator != nil {
+		shopperInteraction, ok := initiatorTypeToShopperInteraction[*authRequest.ProcessingInitiator]
```

Comment:
I think you can do this in one line:
```suggestion
		if shopperInteraction, ok := initiatorTypeToShopperInteraction[*authRequest.ProcessingInitiator]; ok {
```